portal/models/fhir.py

```diff
@@ -526,11 +567,31 @@ def consolidate_answer_pairs(answers):
 
         return filtered_answers
 
+    def entry_method(row_data, qnr_data):
+        # Todo: replace with EC.PAPER CodeableConcept
+        if 'paper' in (c.get('code') for c in qnr_data['encounter']['type']):
+            return 'enter manually - paper'
+        if row_data.get('subject_id') == row_data.get('author_id'):
+            return 'online'
+        else:
+            return 'enter manually - interview assisted'
+
+    def author_role(row_data, qnr_data):
```

Comment:
I don't believe we need `author_role` - this should be covered by the encounter.